### ISSUE TITLE
Drop BCL compilation mode with mcs

### ISSUE THREAD

**Issue by marek-safar** [mono/mono#9388]
We are hitting mcs limitations more frequently than before when importing more CoreFX code and we cannot keep up with csc development. We need to finish the work on fixing Roslyn to work on BE to allow us to drop mcs support for Mono's BCL.

**Comment by marek-safar** [comment]
Roslyn bumped SRM https://github.com/dotnet/roslyn/pull/27023 (finally!)

**Comment by lewurm** [comment]
Roslyn not running on BE is not the only blocker imho. On PowerPC there are still some codegen issues to make roslyn work reliable for example. I assume it's going to be similar for s390x.

**Comment by marek-safar** [comment]
@lewurm yep but that should not be too hard to fix, right? Do we track them anywhere?

**Comment by lewurm** [comment]
@marek-safar I made some progress on ppc64le (= little endian) a few weeks ago. With that it's possible to run roslyn sort of, but it's very crashy and unreliable. No idea idea how much work it's going to need to make it usable. I guess ppc64 big endian will be in the same boot once the roslyn big endian patch lands. I would say let's wait and see how well roslyn works on s390x and then decide what to do.

/cc @nealef

**Comment by directhex** [comment]
I would love to see a release with https://github.com/mono/mono/issues/7445 resolved ship before Roslyn becomes mandatory - Roslyn still isn't in sufficient shape to ship in the downstream distribution repositories, so most distributions are still on an old 4.x release that was still buildable with a completely from-source in-tree stack.
